modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/run_scripts_in_element.es.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+function runJSFromText(text) {
+	const scriptElement = document.createElement('script');
+
+	scriptElement.text = text;
+
+	document.head.appendChild(scriptElement);
+
+	scriptElement.remove();
+
+	return scriptElement;
+}
+
+function runJSFromFile(src) {
+	const scriptElement = document.createElement('script');
+
+	scriptElement.src = src;
+
+	const callback = function () {
+		scriptElement.remove();
+	};
+
+	scriptElement.addEventListener('load', callback);
+	scriptElement.addEventListener('error', callback);
+
+	document.head.appendChild(scriptElement);
+
+	return scriptElement;
+}
+
+function runScript(scriptElement) {
+	if (scriptElement.type && scriptElement.type !== 'text/javascript') {
+		return;
+	}
```

Comment:
Are you sure about this check? Your own `scriptElement` created on line 16 wouldn't pass this check (`type` there would be `''`).

The type [can be omitted](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script), in which case, it's JavaScript.

Reply:
Ignore me, I didn't even notice the bit before the `&&`.